src/fidesops/service/connectors/query_config.py

```diff
@@ -198,10 +192,17 @@ def generate_query(
     def generate_update_stmt(self, row: Row, policy: Policy) -> Optional[TextClause]:
         update_value_map = self.update_value_map(row, policy)
         update_clauses = [f"{k} = :{k}" for k in update_value_map]
-        pk_clauses = [f"{k} = :{k}" for k in self.primary_keys]
+        non_empty_primary_keys = filter_nonempty_values(
+            {
+                f.name: f.cast(row[f.name])
+                for f in self.primary_key_fields
+                if f.name in row
+            }
+        )
+        pk_clauses = [f"{k} = :{k}" for k in non_empty_primary_keys]
 
-        for pk in self.primary_keys:
-            update_value_map[pk] = row[pk]
+        for k, v in non_empty_primary_keys.items():
+            update_value_map[k] = v
```

Comment:
overall can we make the above more readable, ie instead of `f` and `k` using more descriptive var names?